decimalSuite_test.go

```diff
@@ -0,0 +1,127 @@
+
+// TODO: any tests that are failing a particular test in the test suite will be turned into a unit test
+// func TestNew(t *testing.T) {
+// 	// require := require.New(t)
+// 	// // propper rounding
+// 	// require.Equal(MustParseDecimal64("4444444444444445"), MustParseDecimal64("4444444444444444").Add(MustParseDecimal64("0.5001")))
+// 	// require.Equal(MustParseDecimal64("0.23"), MustParseDecimal64("1.3").Add(MustParseDecimal64("-1.07")))
+// 	// fmt.Println("sjoidgf", MustParseDecimal64("4444444444444444").Add(MustParseDecimal64("1.5001")))
+// 	// // require.Equal(MustParseDecimal64("12345678901234.29"), MustParseDecimal64("12345678901234").Add(MustParseDecimal64("0.2951")))
+//
+// }
+
+//getInput gets the test file and extracts test using regex, then returns a map object and a list of test names
+func getInput(file string) (data []testCaseStrings) {
+	dat, _ := ioutil.ReadFile("dectest/ddAdd.decTest")
+	dataString := string(dat)
+	//  "(\n)") // start with newline
+	//  "(?P<TestName>dd[\w]*)") // TestName must start with dd (double decimal)
+	//  "(?P<TestFunc>[\S]*)") // testfunc made of anything that isn't a whitespace
+	//  "(\s*\'?)") // after can be any number of spaces and quotations if they exist
+	// "(?P<TestVals_1>(\+|-)*[^\t\f\v\' ]*)") // first test val is anything that isnt a whitespace or a quoteation mark
+	//  "(?P<TestVals_1>(\+|-)*[^\t\f\v\' ]*)") // first test val is anything that isnt a whitespace or a quoteation mark
+	// ('?\s*'?) match any quotation marks and any spaces
+	// (?P<TestVals_2>(\+|-[^->])?[^\t\f\v\' ]*) testvals2 same as 1 but no '->'
+	// ('?\s*->\s*'?) matches the indicator to answer
+	// (?P<answer>(\+|-)*[^\t\f\v\' ]*) matches the answer that's anything that is plus minus but not quotations
+	// split into groups: testName, TestFunct, TestVals (x2) and TestAns
+	regex := `(\n)(?P<TestName>dd[\w]*)(\s*)(?P<TestFunc>[\S]*)(\s*\'?)(?P<TestVals_1>(\+|-)*[^\t\f\v\' ]*)('?\s*'?)(?P<TestVals_2>(\+|-[^->])?[^\t\f\v\' ]*)('?\s*->\s*'?)(?P<answer>(\+|-)*[^\r\n\t\f\v\' ]*)`
+	r := regexp.MustCompile(regex)
+	ans := r.FindAllStringSubmatch(dataString, -1)
+	var datum testCaseStrings
+	for i := range ans {
+		datum.op = ans[i][4]
+		datum.val1 = ans[i][6]
+		datum.val2 = ans[i][9]
+		datum.expectedResult = ans[i][12]
+		datum.name = ans[i][2]
+		data = append(data, datum)
+
+	}
+	return data
+
+}
+
+//convertToDec64 converts the map object strings to decimal64s
+func convertToDec64(testvals testCaseStrings) (dec64Vals decValContainer) {
+	dec64Vals.val1, dec64Vals.val1Err = ParseDecimal64(testvals.val1)
+	dec64Vals.expectedResult, dec64Vals.expecValErr = ParseDecimal64(testvals.expectedResult)
+	dec64Vals.val2, dec64Vals.val2Err = ParseDecimal64(testvals.val2)
+	return
+}
+
+// TODO: get doTests to run more functions
+//doTests completes the tests and returns a boolean and string on if the test passes
+func doTests(testVals decValContainer, testValStrings testCaseStrings) (testStatus bool, testString string) {
```

Comment:
Booleans should be named as if they are logical propositions (statements of fact). This is because they represent the truth-or-falsity of a statement. It also makes tests read more naturally and reduces the temptation to needlessly add `== true` or `== false` to such tests.

In this case, `testPassed` would be a good choice (`if testPassed …`).